Translate documentation to 🇬🇧 (#34)
europace/kex-angebote-api: README.md

All our examples use variables.

    {
      "query": "...",
      "variables": { ... }
    }

### Error codes

One of the special features in GraphQL is that most errors are not reflected via HTTP error codes.
In many cases you receive a status code 200, even though an error has occurred. These GraphQL errors can be found in the `errors` field of the response body.
More information about error codes can be found [here](https://docs.api.europace.de/privatkredit/graphql/#error-handling).

### HTTP-Status Errors

| Error Code | Message               | Description                     |
|------------|-----------------------|---------------------------------|
| 401        | Unauthorized          | Authentication failed           |
| 403        | Forbidden             | The API client misses a scope   |
| 415        | Unsupported MediaType | The wrong content type was used |

#### GraphQL Errors

| Error Code | Message     | Description                                                                                     |
|------------|-------------|-------------------------------------------------------------------------------------------------|

> | 400        | Bad Request | Request format ist invalid (mandatory fields are missind, wrong parameter names or values, ...) |

Reviewer: ```suggestion
| 400        | Bad Request | Request format is invalid (mandatory fields are missing, wrong parameter names or values, ...) |
```

| 403        | Forbidden   | The authenticated user does not have sufficient rights                                          |
| 409        | Conflict    | The Vorgang was updated after the offer was calculated and before the offer was accepted        |

## Schaufensterkonditionen

Schaufensterkonditionen, both the top Schaufensterkondition and a complete list, can be retrieved via our GraphQL API using **HTTP POST**.  
The URL for the Schaufensterkonditionen is:

    https://kex-angebote.ratenkredit.api.europace.de/schaufenster

### Query Top-Schaufensterkondition

#### Request

The GraphQL-Query is called `topSchaufensterkondition` and contains the following parameter:

| Parameter Name     | Type                                      | Default Value                    |
|--------------------|-------------------------------------------|----------------------------------|
| partnerId          | [Partner-ID](#partner-id)                 | The Partner-ID of the API Client |
| auszahlungsbetrag  | Euro!                                     | - (mandatory field)              | 
| laufzeitInMonaten  | Int                                       | -                                | 

> | finanzierungszweck | [Finanzierungszweck](#finanzierungszweck) | All Finanzierungszwecke          |

Reviewer: ```suggestion
| finanzierungszweck | [Finanzierungszweck](#finanzierungszweck) | All financing purposes          |
```

Reviewer: Also nur als Idee, da es hier erklärt wird. Sehe aber auch ein, wenns es englisch bleiben soll.

Author: Hmm, wir haben das im Glossar stehen (sobald mein PR fertig ist). und hier ist ja im prinzip gemeint, dass alle Werte von Finanzierungszweck genommen werden, daher würde ich das ungern übersetzen.
Was hälst du hiervon?
```suggestion
| finanzierungszweck | [Finanzierungszweck](#finanzierungszweck) | Calculation over all values of Finanzierungszweck          |
```

            ]
        }
    }

## Angebote

You can calculate a list of machbare and vollständige Angebote using data of a Vorgang via our GraphQL-API using **HTTP POST**.
Using the id of the corresponding Angebot you can accept it.  
The URL for calculating and accepting Angebote is:

    https://kex-angebote.ratenkredit.api.europace.de/angebote  

### Query Angebote

#### Hints

> * The calculation of Angebote is only possible, if the corresponding Vorgang includes a valid **Kreditbetrag** and **Laufzeit or Rate**. Should that not be the case  the API user receives a [GraphQL-Error](#graphql-errors) with the status code `400`. The Vorgang has to be corrected before you can continue.

Reviewer: ```suggestion
* The calculation of Angebote is only possible if the corresponding Vorgang includes a valid **Kreditbetrag** and **Laufzeit or Rate**. Should that not be the case the API user receives a [GraphQL-Error](#graphql-errors) with the status code `400`. The Vorgang has to be corrected before you can continue.
```

Reviewer: Ich hätte nur das doppelte Leerzeichen entfernt, Deepl meinte aber auch das Komma wäre da falsch 😅

Author: Susanne scheint Kommas zu mögen, daher hab ich mir das angewöhnt 😅

### Mutation Angebot-Annehmen

#### Hinweise
#### Hints

* Aktuell unterstützt die API nur das Fernabsatzgeschäft.
* Aktuell unterstützt die API nur den Vertriebskanal B2B2C.
* Der authentifizierte Nutzer muss zum Zeitpunkt der Annahme eine Handelsbeziehung für die Bank besitzen, in der die Annahme erlaubt ist.  
  Andernfalls erhält der Nutzer einen [GraphQL-Error](#fehlercodes) mit dem Statuscode `403`
* Das Annehmen von ermittelten Angeboten ist nur möglich, wenn der Vorgang aktuell ist. Sollte nach der Ermittlung und vor der Annahme eine Änderung am Vorgang vorgenommen werden, so erhält der Nutzer der API einen [GraphQL-Error](#fehlercodes) mit dem Statuscode `409`. In diesem Fall ist eine erneute Ermittlung notwendig.
* Zur Optimierung des Angebotsprozess ermitteln wir unter Umständen zusätzliche Alternativangebote unter Adjustierung der Kreditparameter.
* Der im Vorgang eingetragene Kundenbetreuer ist für die Annahme wichtig, da Name und Kontaktdaten an die Banken geschickt werden.

> * Wenn sich das Angebot im Annahmeprozess als nicht machbar herausstellt, wird in der Schnittstelle kein Antrag zurückgegeben.

Author: ich habe das rausgenommen, weil es im Annahmejob schon steht und ich es auch eher dort sehe als hier

### Query Annahme-Job

#### Hints

> * If the offer cannot be accepted from the bank, the response will not contain any Antrag.

Author: ist das dann eigentlich der status "failure" , oder ist das status success und dann kein Antrag? Würde das hier gerne noch ergänzen, weiß aber den Prozess gerade nicht

Reviewer: {
  "data": {
    "annahmeJob": {
      "status": "SUCCESS"
    }
  },
  "errors": []
}

Also status "SUCCESS" und ein kein Antrag kommt zurück, wenn die Anfrage von der Bank abgelehnt wird. Hilft das?

Reviewer: <img width="1039" alt="Bildschirmfoto 2021-12-10 um 15 34 07" src="https://user-images.githubusercontent.com/8448531/145590485-08471ea7-f719-4365-802e-a4cb18e89a7a.png">

Author: ja hilft sehr, danke 👍🏻

#### Request

The GraphQL-Query is called `annahmeJob` and has the following parameter:

| Parameter Name | Type    | Default Value       | Comment                                                                      |
|----------------|---------|---------------------|------------------------------------------------------------------------------|
| jobId          | String! | - (mandatory field) | The ID of the job, which was returned when initiating the acceptance process |

#### Response

> This Query Returns a [AnnahmeJob](#annahmejob). It contains the Status of the acceptance process and the Antrag if the process was successful.

Reviewer: ```suggestion
This Query returns a [AnnahmeJob](#annahmejob). It contains the Status of the acceptance process and the Antrag if the process was successful.
```

Reviewer: Ist es "an" AnnahmeJob?

Author: ```suggestion
This Query returns an [AnnahmeJob](#annahmejob). It contains the Status of the acceptance process and the Antrag if the process was successful.
```

      "errors": []
    }

## Request Datatypes

### Partner-ID

This type is a 5-digit String and identifies a Plakette of the Europace-Partnermanagement.  
The Partner-ID has to be underneath or identical to the Partner-ID of the API-Client.

### Datenkontext

This type is a String which can currelty have the following values
* TESTUMGEBUNG
* ECHTGESCHAEFT

### Finanzierungszweck

> This type is a String which can currelty have the following values

Reviewer: ```suggestion
This type is a String which can currently have one of the following values
```
